src/reader.rs

```diff
@@ -4971,7 +4973,7 @@ fn reader_run_command(parser: &Parser, cmd: &wstr) -> EvalRes {
         );
     }
 
-    term_steal();
+    term_steal(eval_res.status.is_success());
```

Comment:
I'm not yet sure if we should do this but I guess the right criteria is "exited via signal"